pkg/sentry/arch/signal_arm64.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package arch
+
+import (
+	"encoding/binary"
+	"syscall"
+
+	"gvisor.dev/gvisor/pkg/abi/linux"
+	"gvisor.dev/gvisor/pkg/sentry/usermem"
+)
+
+// SignalContext64 is equivalent to struct sigcontext, the type passed as the
+// second argument to signal handlers set by signal(2).
+type SignalContext64 struct {
+	FaultAddr uint64
+	Regs      [31]uint64
+	Sp        uint64
+	Pc        uint64
+	Pstate    uint64
+	_pad      [8]byte // __attribute__((__aligned__(16)))
+	Reserved  [4096]uint8
+}
+
+// UContext64 is equivalent to ucontext on arm64(arch/arm64/include/uapi/asm/ucontext.h).
+type UContext64 struct {
+	Flags  uint64
+	Link   *UContext64
+	Stack  SignalStack
+	Sigset linux.SignalSet
+	/* glibc uses a 1024-bit sigset_t */
+	_pad [16]uint64
```

Comment:
> arch/arm64/include/uapi/asm/ucontext.h:
....
       __u8              __unused[1024 / 8 - sizeof(sigset_t)];

I don't understand why _pad is 128 bytes here...

Reply:
Here _pad is [16]uint64, but in defs_linux_arm64.go _pad is [(1024 - 64) / 8]byte. Why they are different?

Reply:
Hi @avagin 
I combined these two (_pad & _pad2) into a single one (_pad).
In golang:
type ucontext struct {
	uc_flags    uint64
	uc_link     *ucontext
	uc_stack    stackt
	uc_sigmask  uint64
	_pad        [(1024 - 64) / 8]byte
	_pad2       [8]byte // sigcontext must be aligned to 16-byte
	uc_mcontext sigcontext
}

And in gVisor:
type UContext64 struct {
	Flags  uint64
	Link   *UContext64
	Stack  SignalStack
	Sigset linux.SignalSet
	/* glibc uses a 1024-bit sigset_t, and sigcontext must be aligned to 16-byte */
	_pad [16]uint64
	/* last for future expansion */
	MContext SignalContext64
}

Reply:
Thanks!